decaylanguage/dec/dec.py

```diff
@@ -612,6 +620,59 @@ def __str__(self):
         return repr(self)
 
 
+class DecayModelParamValueReplacement(Visitor):
+    """
+    Lark Visitor implementing the replacement of decay model parameter names
+    with the actual parameter values provided in 'Define' statements,
+    and replacement of floats stored as strings to the actual floating values.
+    The replacement is only relevant for Lark Tree instances of name
+    'model_options' (Tree.data == 'model_options').
+
+    Parameters
+    ----------
+    define_defs: dict, optional, default={}
+        Dictionary with the 'Define' definitions in the parsed file.
+        Argument to be passed to the class constructor.
+
+    Examples
+    --------
+    >>> from lark import Tree, Token
+    >>> ...
+    >>> t = Tree('decay', [Tree('particle', [Token('LABEL', 'Upsilon(4S)')]),
+    ...         Tree('decayline', [Tree('value', [Token('SIGNED_NUMBER', '1.0')]),
+    ...         Tree('particle', [Token('LABEL', 'B0')]),
+    ...         Tree('particle', [Token('LABEL', 'anti-B0')]),
+    ...         Tree('model', [Token('MODEL_NAME', 'VSS_BMIX'),
+    ...         Tree('model_options', [Token('LABEL', 'dm')])])])])
+    >>> dict_define_defs = {'dm': 0.507e12}
+    >>> DecayModelParamValueReplacement(define_defs=dict_define_defs).visit(t)
+    Tree(decay, [Tree(particle, [Token(LABEL, 'Upsilon(4S)')]), Tree(decayline,
+    [Tree(value, [Token(SIGNED_NUMBER, '1.0')]), Tree(particle, [Token(LABEL, 'B0')]),
+    Tree(particle, [Token(LABEL, 'anti-B0')]), Tree(model, [Token(MODEL_NAME, 'VSS_BMIX'),
+    Tree(model_options, [Token(LABEL, 507000000000.0)])])])])
+    """
+    def __init__(self, define_defs=dict()):
+        self.define_defs = define_defs
+
+    def _replacement(self, t):
+        try:
+            t.children[0].value = float(t.children[0].value)
+        except AttributeError:
+            try:
+                t.value = self.define_defs[t.value]
```

Comment:
Can you make this an if statement instead of making it a catch all statement?

```python
if t.value in self.define_defs:
    t.value = self.define_defs[t.value]
```

You could also catch just KeyErrors or something like that. `except:` is frowned upon.

Fun fact: Python 3.8+ syntax:
```python
if (val := t.value) in self.define_defs:
    t.value = self.define_defs[val]
```

Reply:
Fair enough.